### PR TITLE
experience/6285: Hookup /validate page with new /validate endpoint

### PR DESCRIPTION
**Pull request by sethdarragile6**
This PR modifies the /validate page in the web app to point to the newly created /validate API endpoint.

Test Steps:
1. Log into RS as prime admin
2. Make sure the "validation-service" feature flag is set
3. Set your org and choose "Validate" from the "Upload & Validate" menu
4. Upload your sample HL7 or CSV files, observing that...
4a. The display of errors, warnings or successes match when uploading same file from "Upload & Validate" -> "User Upload"
4b. No new entries result in the "Submissions" page

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Relateds to #6285 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: frontend-react/src/components/FileHandlers/FileHandler.tsx [hunk 1]
```diff
@@ -228,6 +232,7 @@ const FileHandler = ({
         return <FileHandlerSpinner message="Loading..." />;
     }
 
+    console.log();
```
Comment (ahay-agile6) [inline frontend-react/src/components/FileHandlers/FileHandler.tsx:235]: remove

File: frontend-react/src/components/FileHandlers/FileHandler.test.tsx [hunk 2]
```diff
@@ -181,6 +184,7 @@ describe("FileHandler", () => {
                     showSuccessMetadata={false}
                     showWarningBanner={false}
                     warningText=""
+                    endpointName="waters"
```
Comment (ahay-agile6) [inline frontend-react/src/components/FileHandlers/FileHandler.test.tsx:187]: can we turn this into an enum? "waters" and "validate"?

**Review by ahay-agile6 — CHANGES_REQUESTED** [review]
one `console.log` removal and a suggestion

File: frontend-react/src/pages/Validate.tsx [hunk 3]
```diff
@@ -1,7 +1,8 @@
-import watersApiFunctions from "../network/api/WatersApiFunctions";
 import FileHandler, {
     FileHandlerType,
 } from "../components/FileHandlers/FileHandler";
+// import validateApiFunctions from "../network/api/ValidateApiFunctions";
```
Comment (doug-s-nava) [inline frontend-react/src/pages/Validate.tsx:4]: this can get removed / reverted

**Review by doug-s-nava — APPROVED** [review]
lgtm once you remove a couple of unused things.

one things on the API side though, when I submit a file with no errors, only warnings, it comes back with a 400. Is that expected? The upload endpoint returns a 201 for the same file.

File: frontend-react/src/components/FileHandlers/FileHandler.tsx (hunk 1 above)
Comment (sethdarragile6) [reply frontend-react/src/components/FileHandlers/FileHandler.tsx]: removed

File: frontend-react/src/components/FileHandlers/FileHandler.test.tsx (hunk 2 above)
Comment (sethdarragile6) [reply frontend-react/src/components/FileHandlers/FileHandler.test.tsx]: done!

File: frontend-react/src/pages/Validate.tsx (hunk 3 above)
Comment (sethdarragile6) [reply frontend-react/src/pages/Validate.tsx]: removed!

**Comment by sethdarragile6** [comment]
> lgtm once you remove a couple of unused things.
> 
> one things on the API side though, when I submit a file with no errors, only warnings, it comes back with a 400. Is that expected? The upload endpoint returns a 201 for the same file.

@doug-s-nava  Good call out. I'll have to see what's the haps there on the API side, and follow up with a ticket or fix

File: frontend-react/src/components/FileHandlers/FileHandler.test.tsx [hunk 4]
```diff
@@ -96,6 +100,7 @@ describe("FileHandler", () => {
                 showSuccessMetadata={false}
                 showWarningBanner={false}
                 warningText=""
+                endpointName="waters"
```
Comment (ahay-agile6) [inline frontend-react/src/components/FileHandlers/FileHandler.test.tsx:103]: can use the enum here, too!

File: frontend-react/src/components/FileHandlers/FileHandler.test.tsx [hunk 5]
```diff
@@ -122,6 +127,7 @@ describe("FileHandler", () => {
                     showSuccessMetadata={false}
                     showWarningBanner={false}
                     warningText=""
+                    endpointName="waters"
```
Comment (ahay-agile6) [inline frontend-react/src/components/FileHandlers/FileHandler.test.tsx:130]: can use the enum here, too!

File: frontend-react/src/components/FileHandlers/FileHandler.test.tsx [hunk 6]
```diff
@@ -154,6 +160,7 @@ describe("FileHandler", () => {
...
```
Comment (ahay-agile6) [inline frontend-react/src/components/FileHandlers/FileHandler.test.tsx:163]: can use the enum here, too!

**Review by ahay-agile6 — CHANGES_REQUESTED** [review]

**Review by ahay-agile6 — APPROVED** [review]
LGTM! :ship:

File: frontend-react/src/components/FileHandlers/FileHandler.test.tsx (hunk 6 above)
Comment (sethdarragile6) [reply frontend-react/src/components/FileHandlers/FileHandler.test.tsx]: got it (them). thanks!

**Pull request merged**